src/wp-admin/users.php

```diff
@@ -250,7 +250,7 @@
 
 			// Send the password reset link.
 			$user = get_userdata( $id );
-			if ( retrieve_password( $user->user_login ) ) {
+			if ( true === retrieve_password( $user->user_login ) ) {
```

Comment:
I think this is the only change that's needed in this PR. It directly addresses the issue the ticket highlighted and I think we can discuss and reach consensus on consistency with these checks in a separate ticket.

Reply:
Pinging @audrasjb and @SergeyBiryukov for their thoughts on whether we should keep this PR specific to resolving the bug, and separate out code quality improvements as they likely need some discussion to reach consensus.

Reply:
@petitphp @costdev I've got two opposite comments. So, should I or should I not edit other lines than users.php:253?

p.s. I prefer to do that, so code quality would improve a bit

Reply:
Hello 👋 

I agree with @costdev that only this fix is needed. I explained it in [more details in the Trac ticket](https://core.trac.wordpress.org/ticket/58407#comment:35).

Thank you for your contributions!

Reply:
I'll close this PR when the change is committed. So no additional changes are needed in this PR.